rust/types/src/api_types.rs

```diff
@@ -0,0 +1,13 @@
+use uuid::Uuid;
+
+#[derive(Clone)]
+pub struct CreateDatabaseRequest {
+    pub database_id: Uuid,
+    pub tenant_id: String,
+    pub database_name: String,
+}
+
+#[derive(Clone)]
+pub struct CreateDatabaseResponse {
+    pub success: bool,
```

Comment:
This feels a bit off to me, do we really want this here? Should the response type in rust be more rust-y? 

Should our errors also go in to this definition? That seems nice.